src/database/hooks/useSaveAnonChatHook.ts

```diff
@@ -65,34 +68,44 @@ const useSaveAnonChatHook = () => {
               member,
               initAnonymousChat?.message?.cid
             );
-
             await userMember.saveOrUpdateIfExists(localDb);
           } catch (e) {
-            console.log('error saveChatFromOtherProfile userMember', e);
+            console.log(e);
           }
           try {
             const memberSchema = ChannelListMemberSchema.fromInitAnonymousChatAPI(
               initAnonymousChat?.message?.cid,
               uuid(),
               member
             );
-            memberSchema.saveIfNotExist(localDb);
+            await memberSchema.saveIfNotExist(localDb);
           } catch (e) {
-            console.log('error saveChatFromOtherProfile memberSchema', e);
+            console.log(e);
           }
         };
 
         Promise.all([saveUserAndChannelListMember()]);
       });
     } catch (e) {
-      console.log('error saveChatFromOtherProfile');
       console.log(e);
     }
-
-    refresh('channelList');
-    refresh('chat');
-
-    if (withNavigate) helperGoToAnonymousChat(object, GROUP_INFO);
+    try {
+      const channelListData = ChannelListSchema.fromInitAnonymousChatAPI(
+        {
+          ...initAnonymousChat
+        },
+        type === SIGNED ? 'PM' : 'ANON_PM'
+      );
+      await channelListData.saveIfLatest(localDb);
+    } catch (e) {
+      console.log(e);
+    } finally {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This is a good use of try-catch-finally. The code attempts to save `channelListData`, logs any errors if they occur, and then refreshes the 'channelList' and 'chat' regardless of whether an error occurred. This ensures that the UI will update even if saving the data fails.


<!-- This is an auto-generated comment by OSS CodeRabbit -->